packages/dd-trace/src/appsec/rule_manager.js

```diff
@@ -26,137 +20,66 @@ function loadRules (config) {
 }
 
 function updateWafFromRC ({ toUnapply, toApply, toModify }) {
-  const batch = new Set()
-
-  const newRulesData = new SpyMap(appliedRulesData)
-  let newRuleset
-  let newRulesetId
-  const newRulesOverride = new SpyMap(appliedRulesOverride)
-  const newExclusions = new SpyMap(appliedExclusions)
-  const newCustomRules = new SpyMap(appliedCustomRules)
   const newActions = new SpyMap(appliedActions)
 
+  let wafUpdated = false
+  let wafUpdatedSuccess = true
+
   for (const item of toUnapply) {
-    const { product, id } = item
+    if (!['ASM_DD', 'ASM_DATA', 'ASM'].includes(item.product)) continue
```

Comment:
Right now this array is created in each loop iteration. Maybe move it outside of the method and make it a set so that there is only a single lookup happening per iteration? :)

```suggestion
    if (PRODUCTS.has(item.product)) continue
```

The same applies in other places.

Reply:
Suggestion has been applied.